### PR TITLE
POM-14 Add material aid search form - pomoc materialna

### PR DESCRIPTION
**Pull request by neu5**

### CODE REVIEW

**Review by sfc-gh-krzymkowski — APPROVED** [review]

File: src/app/find-help/material-support-search/material-support-search-form/material-support-search-form.component.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<form (ngSubmit)="search.emit(data)" #form="ngForm">
+  <div class="container">
+    <div class="row">
+      <div class="col-md-4">
+        <label
+          ><b>{{ "TYPE_OF_SUPPORT" | translate }}</b></label
+        >
+        <mat-form-field appearance="outline">
+          <mat-select placeholder="Wybierz z listy" [(ngModel)]="categories" name="kategoriePomocy">
```
Comment (mgol) [inline src/app/find-help/material-support-search/material-support-search-form/material-support-search-form.component.html:9]: `ngModel` needs to point to the selected value, not all of the values. This is likely what's causing the error as Angular tries to set the value to the whole collection and there's a mismatch.

File: src/app/find-help/material-support-search/material-support-search-form/material-support-search-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+interface Option {
+  code: string;
+  label: string;
+}
+@Component({
+  selector: 'app-material-support-search-form',
+  templateUrl: './material-support-search-form.component.html',
+  styleUrls: ['./material-support-search-form.component.scss'],
+})
+export class MaterialSupportSearchFormComponent {
```
Comment (mgol) [inline src/app/find-help/material-support-search/material-support-search-form/material-support-search-form.component.ts:25]: Please name all the files, folders, types, classes, etc. using `material-aid`, not `material-support`, for consistency with Open API (services, endpoints, types are all named like that).

File: src/app/find-help/material-support-search/material-support-search-form/material-support-search-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { Component, Output, EventEmitter } from '@angular/core';
+import { MaterialAidOffer } from '../../../../api/model/materialAidOffer';
+
+const categories = Object.entries(MaterialAidOffer.CategoryEnum).map(([key, value]) => ({
+  code: key,
+  label: value,
+}));
+
+export interface MaterialSupportQuery {
```
Comment (mgol) [inline src/app/find-help/material-support-search/material-support-search-form/material-support-search-form.component.ts:9]: Delete this type, use `MaterialAidOfferSearchCriteria` from the `api` folder directly.

**Review by mgol — CHANGES_REQUESTED** [review]

File: src/app/find-help/material-aid-search/material-aid-search-form/material-aid-search-form.component.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          <mat-select placeholder="Wybierz z listy" [(ngModel)]="data.category" name="kategoriePomocy">
+            <mat-option *ngFor="let cat of categories" [value]="cat.value">{{ cat.value | translate }}</mat-option>
+          </mat-select>
+        </mat-form-field>
+      </div>
+      <div class="col-md-4">
+        <app-cities-search
+          [(ngModel)]="data.location"
+          name="location"
+          [placeholder]="'Np. Rzeszów, podkarpackie'"
```
Comment (mgol) [inline src/app/find-help/material-aid-search/material-aid-search-form/material-aid-search-form.component.html:18]: BTW, we should also translate placeholders. This one is repeated in a few places so it's fine to leave it for now but we'll need to get to it.

File: src/app/find-help/material-aid-search/material-aid-search.component.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  search(searchCriteria: MaterialAidOfferSearchCriteria) {
+    this.loading = true;
+
+    const pageRequest: Pageable = {
+      // page?: number;
+      // size?: number;
+      // sort?: Array<string>;
+    };
+
+    // TODO a proper request
```
Comment (mgol) [inline src/app/find-help/material-aid-search/material-aid-search.component.ts:25]: This comment is out of date, I think. ;)

File: src/app/find-help/material-aid-search/material-aid-search.component.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import { Component } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
```
Comment (mgol) [inline src/app/find-help/material-aid-search/material-aid-search.component.ts:2]: This is not needed anymore.

File: src/app/find-help/material-aid-search/material-aid-search.component.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+@Component({
+  selector: 'app-material-aid-search',
+  templateUrl: './material-aid-search.component.html',
+  styleUrls: ['./material-aid-search.component.scss'],
+})
+export class MaterialAidSearchComponent {
+  results: MaterialAidOffer[] = [];
+  total?: number = undefined;
+  loading = false;
+  constructor(private http: HttpClient, private materialAidResourceService: MaterialAidResourceService) {}
```
Comment (mgol) [inline src/app/find-help/material-aid-search/material-aid-search.component.ts:14]: Remove the `http` param.

**Review by mgol — CHANGES_REQUESTED** [review]
A few more comments

**Review by mgol — APPROVED** [review]

**Review by sfc-gh-krzymkowski — APPROVED** [review]

**Pull request merged**